### ISSUE TITLE
Continous integration for .devcontainers

### ISSUE THREAD

**Issue by bhack** [microsoft/vscode-remote-release#2570]
How we can test a `.devcontainer` in Continous Integration? Is there a way to pilot  the build and run test code from a Vscode image on a `.devcontainer` with cmdline?
Partially related to cmdline https://github.com/microsoft/vscode-remote-release/issues/2133 but this Is for testing in CI/GitHub.

**Comment by Chuxel** [comment]
//cc: @chrmarti @egamma as FYI

**Comment by bhack** [comment]
> That said, it's an interesting use case that I can see making sense for major project. I'm moving this to the main vscode-remote-release repo, since this would be a feature request for the extension and potentially VS Code itself.
I'am most interested in this last point. Probably we need `code` cmdline control and some kind of Docker in Docker solution to pilot the extension for tests.

**Comment by chrmarti** [comment]
We could look into using the existing integration test infrastructure of VS Code for opening a folder in a container.

We have some internal infrastructure in the extension for our own tests, but we currently don't bundle or expose any of it. This might offer an alternate approach. One advantage of that for our own tests is that it runs without the VS Code UI.

**Comment by bhack** [comment]
@chrmarti Yes without UX is Ideal for CI but  I think that we still need Docker in Docker cause in third party projects we need to run that tool in Docker and then the container of .devcontainer (build, run) rely on Docker.

**Comment by chrmarti** [comment]
If we had an entrypoint on the extension that can run without UI, you could run it in Docker-in-Docker. We currently don't use Docker-in-Docker for our tests, but I don't think there is anything preventing that.

**Comment by bhack** [comment]
@chrmarti Yes was just to have the tool (`code`?) easily available in the container that It Is also running other tests. I think it Is a quite common scenario.
